presto-hive/src/main/java/com/facebook/presto/hive/HiveClientConfig.java

```diff
@@ -604,6 +631,19 @@ public HiveClientConfig setInsertOverwriteImmutablePartitionEnabled(boolean inse
         return this;
     }
 
+    public InsertExistingPartitionsBehavior getInsertExistingPartitionsBehavior()
+    {
+        return insertExistingPartitionsBehavior.orElse(immutablePartitions ? (isInsertOverwriteImmutablePartitionEnabled() ? OVERWRITE : ERROR) : APPEND);
```

Comment:
Let's fail hard if the two configs do not match with each other instead of relying on optional empty

Reply:
If these configs don't match, we will throw error at line 628 checkArgument call. I have removed optional though.